draft-ietf-quic-qlog-main-schema.md

```diff
@@ -659,68 +666,186 @@ Example qlog event:
 
 ## Timestamps {#time-based-fields}
 
-An event's "time" field indicates the timestamp at which the event occurred. Its value is
-typically the Unix timestamp since the 1970 epoch (number of milliseconds since
-midnight UTC, January 1, 1970, ignoring leap seconds). However, qlog supports two
-more succinct timestamps formats to allow reducing file size. The employed format
-is indicated in the "time_format" field, which allows one of three values:
-"absolute", "delta" or "relative".
+Each event MUST include a "time" field to indicate the timestamp that it
+occurred. It is a duration measured from some point in time, in units of ???.
+There are several options for generating and logging timestamps, these are
+governed by the ReferenceTime type (optionally included in the "reference_time"
+field contained in a trace's "common_fields" ({{common-fields}})) and TimeFormat
+type (optionally included in the "time_format" field contained in the event
+itself, or a trace's "common_fields").
+
+There is no requirement that events in the same trace use the same time format.
+However, using a single time format for related events can make them easier to
+analyze.
+
+The reference time is defined as:
 
 ~~~ cddl
-TimeFormat = "absolute" /
-             "delta" /
-             "relative"
+ReferenceTime = {
+    clock_type: "system" / "monotonic" / text .default "system"
+    epoch: RFC3339DateTime / "unknown" .default "1970-01-01T00:00:00.000Z"
+
+    ? wall_clock_time: RFC3339DateTime
+}
+
+RFC3339DateTime = text
```

Comment:
```suggestion
; RFC 3339 UTC string at millisecond resolution. i.e., YYYY-MM-DDTHH:mm:ss.sssZ
RFC3339DateTime = text
```

Reply:
I don't think we need this, its profiling the RFC even further.